src/gui/qgsmaplayerstylecategoriesmodel.cpp

```diff
@@ -65,6 +65,16 @@ void QgsMapLayerStyleCategoriesModel::setCategories( QgsMapLayer::StyleCategorie
   if ( mCategories == categories )
     return;
 
+  // determine the allowed categories to consider only these in the last categories.
+  QgsMapLayer::StyleCategories allowedCategories;
+  for ( const QgsMapLayer::StyleCategory &category : mCategoryList )
```

Comment:
```suggestion
  for ( QgsMapLayer::StyleCategory category : std::as_const( mCategoryList ) )
```

enums can be copied by value cheap enough, no need to reference them